### PR TITLE
feat(core): Make ConfigModule importable in every NgModule

### PR DESCRIPTION
**Pull request by UserGalileo**
Removed the check in the constructor, this way every module can import ConfigModule and have access to ConfigPipe.
Updated ReadME with a one-liner instruction for ConfigPipe.

** PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/fulls1z3/ngx-config/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

** PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

** What is the current behavior?

Issue Number: #149 

** What is the new behavior?

Now every NgModule can import ConfigModule.

** Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

** Other information

I don't think tests are needed for this change: can you confirm?

### CODE REVIEW

**Review by fulls1z3 — CHANGES_REQUESTED** [review]
i appreciate the efforts to make `ConfigPipe` available in every module, but I'd recommend to keep the root check and come up with `forChildren` method.

this way we can both preserve exiting functionality and add new ones.

**Comment by UserGalileo** [comment]
@fulls1z3 Can you confirm that the method should be named `forChildren`? I don't think I've ever seen a `forChildren` method in other libraries, I'd suggest to choose between these 2 options:
- `forFeature` a-la NgRx
- `forChild` a-la Angular Router

But if you choose `forChildren` no problem, I just want to be sure before making new commits :)

Also, for the root check: it's doable but we'd have to perform the check outside of the constructor, just like `Angular Router` does: [this way.
](https://github.com/angular/angular/blob/master/packages/router/src/router_module.ts#L197)
Do you agree with adopting this solution?

**Comment by fulls1z3** [comment]
@UserGalileo I agree and it's a typo - let's follow the Angular pattern and call it `forChild`.

**Comment by fulls1z3** [comment]
@UserGalileo also 👍 for the root guard

**Comment by UserGalileo** [comment]
@fulls1z3 I made the changes on my branch, could you verify if everything works fine?

I'm sorry, I would be happy to provide some more tests for the lazy-loaded modules scenario, but I'm really busy these days.

**Comment by fulls1z3** [comment]
@UserGalileo just fixed the CI pipeline, but it turns out your PR has to pass validation checks such as TSLint.

FYI, you can add the following block to `tslint.json` between lines 4 and 5 - in order that `void` type works for `InjectionToken`.

```

    "invalid-void": [
      true,
      {
        "allow-generics": [
          "EventEmitter",
          "InjectionToken",
          "Observable",
          "Promise"
        ]
      }
    ],
```

File: packages/@ngx-config/core/src/config.module.ts [hunk 1]
```diff
@@ -13,17 +13,37 @@ export const initializerFactory = (config: ConfigService) => {
   return res;
 };
 
+export const CONFIG_FORROOT_GUARD = new InjectionToken<void>('CONFIG_FORROOT_GUARD');
+
+/**
+ * Used to check if the user has called ConfigModule.forRoot() in a lazy-loaded module.
+ */
```
Comment (fulls1z3) [inline packages/@ngx-config/core/src/config.module.ts]: your code is already good - explaining itself. no need for comments ;)

File: packages/@ngx-config/core/src/config.module.ts [hunk 2]
```diff
@@ -13,17 +13,37 @@ export const initializerFactory = (config: ConfigService) => {
...
 @NgModule({
   declarations: [ConfigPipe],
   exports: [ConfigPipe]
 })
 export class ConfigModule {
+  
+  /**
+   * By injecting the guard we're making sure that the check will be made every time
+   * a new ConfigModule is imported, both in the root injector and in lazy loaded modules.
+   */
```
Comment (fulls1z3) [inline packages/@ngx-config/core/src/config.module.ts]: same as above - your code is already crystal clear

**Review by fulls1z3 — CHANGES_REQUESTED** [review]
all clear, even more 🙂

**Comment by UserGalileo** [comment]
Perfect, is there anything left to do on my side? :)

**Comment by fulls1z3** [comment]
yeah, you could follow the code review (removing comments from the code) as well as Sider must return no issues (tslint). Please consider my remarks at https://github.com/fulls1z3/ngx-config/pull/157#issuecomment-560753719 for this.

**Comment by miguelchico** [comment]
Hi @UserGalileo, @fulls1z3 do you expect any progress on this PR?

**Comment by UserGalileo** [comment]
Sorry for the delay, these are really busy months for me. I just had the time to fix the tslint errors, it seems fine now, could you check?

I preferred to disable tslint rules for 2 lines of code instead of messing around with the tslint.json file, since there were only 2 occasions, and I removed the generic from `InjectionToken` since it was useless.

**Review by fulls1z3 — APPROVED** [review]

**Pull request merged**